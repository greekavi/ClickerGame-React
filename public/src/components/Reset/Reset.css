#button2{
    background-color: #62055f;
  
    color: white;
}
#button2:hover{
    background-color:#f688f7 ;
    color:   #62055f;
}