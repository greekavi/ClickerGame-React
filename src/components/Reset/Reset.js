import React,{useState} from "react";
import Button from '@material-ui/core/Button';
import './Reset.css'
function Reset({ResetGame,ResetEnable}){  
    return(
<div>
<h4>Number of Games played: {ResetGame}</h4>
            <button id="button2" onClick={ResetEnable}>Reset</button>
            
      </div>
    );
}
export default Reset;