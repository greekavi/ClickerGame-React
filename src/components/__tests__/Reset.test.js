import React from 'react';
import ReactDom from 'react-dom';
import {render,fireEvent,cleanup,screen,container} from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "@testing-library/jest-dom/extend-expect";
import {shallow,configure} from 'enzyme';
import Reset from './../Reset/Reset';
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() })

afterEach(() => {
    jest.clearAllMocks();
  });

  it("Check if Reset button works",()=>{
    const ResetEnable=jest.fn();
    const wrapper=shallow(<button id="button2" onClick={ResetEnable}>Reset</button>);
    wrapper.find("#button2").simulate("click");
    expect(ResetEnable.mock.calls.length).toEqual(1);
  })


  it("Passes variables correctly from counter to reset",()=>{
    const {getByTestId}=render(<Reset ResetGame="10"/>);
    expect(getByTestId('resetGameId')).toHaveTextContent("10");
})

it("Matches Snapshot",()=>{
    const tree=renderer.create(<Reset ResetGame="10"/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

