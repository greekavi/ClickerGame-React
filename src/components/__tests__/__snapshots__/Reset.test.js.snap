// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div>
  <h4
    data-testid="resetGameId"
  >
    Number of Games played: 
    10
  </h4>
  <button
    id="button2"
  >
    Reset
  </button>
</div>
`;
